### PR TITLE
add err msg on PeeringRead not found

### PR DESCRIPTION
**Pull request by FFMMM**
### Description
Came across this as a convenience when working with nitya; would be nice to have!

### Testing & Reproduction steps

* updated tests

### CODE REVIEW

**Review by DanStough — APPROVED** [review]
lgtm!

**Review by eculver — APPROVED** [review]
Yeah, this does seem handy. I do always wonder about information leakage when it comes to 404s so we might want to get another set of eyes on it to make sure we aren't sharing more info than we should be in these cases. I wonder if @rboyer @freddygv @banks @mkeeler would have thoughts?

File: agent/peering_endpoint.go
Comment (freddygv) [inline agent/peering_endpoint.go:51]: I would prefer if we used the same error convention that we do for partitions/namespaces:
```suggestion
		return nil, HTTPError{StatusCode: http.StatusNotFound, Reason: fmt.Sprintf("Peering not found for %q", name)}
```

**Comment by mkeeler** [comment]
FWIW, this PR does not leak any new information. The name output in the error reason was user supplied input and they could have inferred as much from just a 404. As for 404s, once we have ACLs for looking up peering information we should:

1. Resolve the token.
2. Authorize the request and return a 403 if needed
3. Lookup the peer
4. Return info or return a 404.

That way if you wouldn't be authorized to view the peer we 403 early and don't let you know whether it exists or not.

**Review by mkeeler — APPROVED** [review]

File: agent/peering_endpoint_test.go
Comment (mkeeler) [inline agent/peering_endpoint_test.go:292]: I think you need to update this to match the change to the message.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/668069.

**Pull request merged**
